### PR TITLE
doc(ISSUE_TEMPLATE): use `.yml` format instead of `.md` (close #9062)

### LINKED ISSUE #9062: Updates : New issue templates using .yml instead of .md 🛠
### Is your proposal related to a problem?

Currently we are using `.md` files instead of the brand new issue forms with `.yml` files which makes it more easier to fill in the data and raise an issue. Using the `.md` files can result in folks skipping out important details while making an issue.

### Describe the solution you'd like

The solution is to build a new `.yml` file which can be customized. Adding an issue Template allows you to organize the issues made by folks in a much better way.
 - We can know if someone wants to write a test or maybe fix something in the website
 - The issue template forms makes writing details more efficient and quick
 - Also since there are `required` options, you can never miss on important values !

 - Here's a general idea : 

image

image

### Describe alternatives you've considered

N/A

### If the feature is approved, would you be willing to submit a PR?

YES ! ,  I'd love to contribute to the project !

### PR DESCRIPTION
**Pull request by progfay**
### Description

In this PR, I converted issue templates from `.md` to `.yml`.

### Changelog

No changes which user affected.

#### Short Changelog

use yml format on issue templates

#### Long Changelog

define issue form with YAML issue templates

### Related Issues

https://github.com/hasura/graphql-engine/issues/9062

### Steps to test and verify

You can check the content and design of issue form from below links:
- Bug report: https://github.com/progfay/hasura-graphql-engine/issues/new?assignees=&labels=k%2Fbug&template=01_bug_report.yml&title=%5BBug%5D%3A+
- Feature request: https://github.com/progfay/hasura-graphql-engine/issues/new?assignees=&labels=k%2Fenhancement&template=02_feature_request.yml&title=%5BRequest%5D%3A+

#### Catalog upgrade

Does this PR change Hasura Catalog version?
- [x] No
- [ ] Yes
  - [ ] Updated docs with SQL for downgrading the catalog 

#### Metadata

Does this PR add a new Metadata feature?
- [x] No
- [ ] Yes
  - Does `run_sql` auto manages the new metadata through schema diffing?
    - [ ] Yes
    - [ ] Not required
  - Does `run_sql` auto manages the definitions of metadata on renaming?
    - [ ] Yes
    - [ ] Not required
  - Does `export_metadata`/`replace_metadata` supports the new metadata added?
    - [ ] Yes
    - [ ] Not required

#### GraphQL
- [x] No new GraphQL schema is generated
- [ ] New GraphQL schema is being generated:
   - [ ] New types and typenames are correlated
   
   

#### Breaking changes

- [x] No Breaking changes
- [ ] There are breaking changes:

  1. Metadata API

     Existing `query` types:
     - [ ] Modify `args` payload which is not backward compatible
     - [ ] Behavioural change of the API
     - [ ] Change in response `JSON` schema
     - [ ] Change in error code
     

  2. GraphQL API

     Schema Generation:
     
     
     - [ ] Change in any `NamedType`
     - [ ] Change in table field names
     

     Schema Resolve:-
     
     - [ ] Change in treatment of `null` value for any input fields 
     

  3. Logging

     - [ ] Log `JSON` schema has changed
     - [ ] Log `type` names have changed

### CODE REVIEW

**Comment by progfay** [comment]
Hello, @IAmTamal and @adas98012.

I converted Issue Templates from `.md` to `.yml`, please review my PR.

Thanks!

**Comment by IAmTamal** [comment]
> Hello, @IAmTamal and @adas98012.
> 
> I converted Issue Templates from `.md` to `.yml`, please review my PR.
> 
> Thanks!

Now the thing is i was waiting to work on it but all of a sudden you made a PR
I was waiting as i was not assigned nor did i know if they even wanted these changes.

File: .github/ISSUE_TEMPLATE/01_bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    attributes:
+      label: "CLI Version"
+      description: "for CLI related issue"
+    validations:
+      required: false
+  - type: dropdown
+    id: environment
+    attributes:
+      label: "Environment"
+      description: "Provide a clear description of what is the current behaviour."
```
Comment (dsandip) [inline .github/ISSUE_TEMPLATE/01_bug_report.yml:23]:
```suggestion
      description: "Where is the issue being seen"
```

File: .github/ISSUE_TEMPLATE/01_bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      options:
+        - Cloud
+        - OSS
+        - EE
+    validations:
+      required: true
+  - type: textarea
```
Comment (dsandip) [inline .github/ISSUE_TEMPLATE/01_bug_report.yml:30]:
```suggestion
  - type: textarea
    id: current-behaviour
    attributes:
      label: "Current Behaviour"
      description: "What is the current behaviour?"
    validations:
      required: false
  - type: textarea
```

File: .github/ISSUE_TEMPLATE/01_bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    id: expected-behaviour
+    attributes:
+      label: "What is the expected behaviour?"
+      description: "Provide a clear description of what you want to happen."
+    validations:
+      required: true
```
Comment (dsandip) [inline .github/ISSUE_TEMPLATE/01_bug_report.yml:36]:
```suggestion
      required: false
```

File: .github/ISSUE_TEMPLATE/01_bug_report.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  - type: textarea
+    id: reproduce
+    attributes:
+      label: "How to reproduce the issue?"
+      value: |
+        1.
+        2.
+        3.
+    validations:
+      required: true
```
Comment (dsandip) [inline .github/ISSUE_TEMPLATE/01_bug_report.yml:46]:
```suggestion
      required: false
```

File: .github/ISSUE_TEMPLATE/01_bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    validations:
+      required: false
+  - type: textarea
+    id: problem-location
+    attributes:
+      label: "Can you identify the location in the source code where the problem exists?"
+      description: "Provide your answer here."
+    validations:
+      required: false
+  - type: textarea
```
Comment (dsandip) [inline .github/ISSUE_TEMPLATE/01_bug_report.yml:75]:
```suggestion
  - type: input
```

File: .github/ISSUE_TEMPLATE/01_bug_report.yml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    id: pr
+    attributes:
+      label: "If the bug is confirmed, would you be willing to submit a PR?"
+      description: "Provide your answer here."
+    validations:
+      required: false
+  - type: textarea
```
Comment (dsandip) [inline .github/ISSUE_TEMPLATE/01_bug_report.yml:82]:
```suggestion
  - type: input
```

File: .github/ISSUE_TEMPLATE/02_feature_request.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+labels: ["k/enhancement"]
+body:
+  - type: textarea
+    id: proposal
+    attributes:
+      label: "Is your proposal related to a problem?"
+      description: "Provide a clear and concise description of what you want to happen."
+    validations:
+      required: true
+  - type: textarea
```
Comment (dsandip) [inline .github/ISSUE_TEMPLATE/02_feature_request.yml:13]:
```suggestion
  - type: textarea
    id: solution
    attributes:
      label: "Describe the solution you'd like"
      description: "Provide a clear and concise description of what you want to happen."
    validations:
      required: false
  - type: textarea
```

File: .github/ISSUE_TEMPLATE/02_feature_request.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    id: alternative
+    attributes:
+      label: "Describe alternatives you've considered"
+      description: "Provide your answer here."
+    validations:
+      required: false
+  - type: textarea
```
Comment (dsandip) [inline .github/ISSUE_TEMPLATE/02_feature_request.yml:20]:
```suggestion
  - type: input
```

**Review by dsandip — CHANGES_REQUESTED** [review]

**Comment by Stefmore02** [comment]
Hey! @progfay 

Here's a friendly reminder to address the changes requested. Let us know if any questions pop out! 🤓

**Comment by Stefmore02** [comment]
Hey! @progfay 

Closing this PR due to inactivity. I encourage you to re-open it whenever you're ready to address the changes 🤓.

**Comment by IAmTamal** [comment]
> Hey! @progfay
> 
> Closing this PR due to inactivity. I encourage you to re-open it whenever you're ready to address the changes nerd_face.

Should i make the changes and make a PR , i originally opened this issue a long time back.

**Pull request closed without merging**